### ISSUE TITLE
Specify prefix for SchemaVerions.ScriptName

### ISSUE THREAD

**Issue by nmummau** [drwatson1/dbup-cli#6]
This is more of a feature request than an actual issue with the tool.

I am on a team that uses DbUp like this: We have a C# DbUp project called "DatabaseDeploy". Within that project are script folders for each development sprint - something like "Sprint0001", and "Sprint0002", etc. Each sprint we add a new sprint folder and add scripts to it.

When using DbUp normally thru that project, the result in the SchemaVersions.ScriptName table are inserted rows something like this:
DatabaseDeploy.Sprint0001.Script-000010-do the thing.sql
DatabaseDeploy.Sprint0001.Script-000020-do the thing.sql
DatabaseDeploy.Sprint0002.Script-000010-do the thing.sql
DatabaseDeploy.Sprint0002.Script-000020-do the thing.sql

Take note that the "DatabaseDeploy" project name, and the folder name "Sprint0001" is prefixed to the name of the .sql file name.

However, when using the dbup-cli, the SchemaVersions.ScriptName is something different, even when pointing to the same folders:
Script-000010-do the thing.sql
Script-000020-do the thing.sql
Script-000010-do the thing.sql
Script-000020-do the thing.sql

I'm trying to use your dbup-cli in **parallel** with this existing C# process. This would allow me to either run the dbup-cli (fancy!) without having to open up VisualStudio and manually execute the DbUp project. However that is not possible since the dbup-cli does not use the same naming convention for the script name. 

Any thoughts on how to use the dbup-cli in this scenario?

**Comment by drwatson1** [comment]
This is interesting, I didn't think about such scenarios. Thank you for the idea!

If I understand you right, you have a folder structure like this:

```
DatabaseDeploy
  - Sprint0001
      - Script-000010-do the thing.sql
      - Script-000020-do the thing.sql
   - Sprint0002
      - Script-000010-do the thing.sql
      - Script-000020-do the thing.sql
  - dbup.yml
```

Your `dbup.yml` contains something like:
```yaml
# ... other options are omitted
  scripts:
    -   folder:  .
        subFolders: yes
```

Here you get many scripts with the same name, so using folder names as script name prefix would solve the problem. Do you want to have the same naming conversion as the original `dbup` does?  For example, will it be enough to use only subfolders' names?

**Comment by nmummau** [comment]
What you describe as the folder structure is correct. 
Your question about my `dbup.yml` requires me to edit what I said in the original post  -- when setting up `dbup.yml` like you just suggested (which I believe I had done exactly like that when trying to get it to work), results in SchemaVersions.ScriptName like this:
```
Sprint0183.Script-000010-select users.sql
```
Which means using the `subFolders: yes` property does work to get the folder name included in the SchemaVersions.ScriptName, however the DbUp project name is still missing. 

In my case it needs to have the same naming convention as the original `dbup`. Simply because I have a requirement to use both `dbup` and `dbup-cli` in parallel. Meaning, I could add/execute scripts 10 thru 20 by using `dbup` and then create additional scripts 30 thru 40 and then execute them by using `dbup-cli`. If `dbup-cli` does not use the same naming convention, it would detect scripts 10 thru 20 need to be executed again, and resulting in `dbup-cli` executing all scripts 10 thru 40. When in reality `dbup-cli` should have only run the 2 new scripts, 30 thru 40.

Without knowing the technical details, it seems it would be as simple as adding a "prefixString" property in `dbup.yml` that would allow you to put whatever in there that you need to match the DbUp naming convention.

**Comment by nmummau** [comment]
Consider [DbUp issue 166](https://github.com/DbUp/DbUp/issues/166). What I'm requesting here for the `dbup-cli` is essentially the exact opposite of that request. I would like there to be a way to _add_ the namespace when `dbup-cli` runs the scripts, so that it matches the way DbUp does it normally.

**Comment by drwatson1** [comment]
I plan to add script naming options to resolve the issue:

```yaml
  naming:
    useOnlyFileName: no
    includeBaseFolderName: no
    prefix:
``` 

Setting the `includeBaseFolderName` to `yes`  adds folder name as a script name prefix. Another option is to set prefix as you wish with a `prefix` option. All options can be used in combination.

What do you think? Would it work for you?

**Comment by nmummau** [comment]
Yes this sounds like a perfect solution.

**Comment by drwatson1** [comment]
I've implemented the feature and release a new version next weekend.

**Comment by drwatson1** [comment]
I released a new version [v1.4.0](https://github.com/drwatson1/dbup-cli/releases/tag/v1.4.0) with a requested feature. Feel free to reopen the issue in case of any problems or create a new issue.
